pypeclub/OpenPype
# Deadline: publishing of animation and pointcache on a farm

## Brief description
Some publishes (or extractors) might be too long to run in on a artist machine, so it should be offloaded to a farm.

## Description
Farm runs a host plugin which triggers new python script. This script does pyblish publish process end-to-end.

Animation and pointcache instances have now new 'Farm' field which if set to True, instance will be sent to DL, collected, validated, extracted and integrated there.

## Additional info
THIS PR IS OPEN FOR DISCUSSION

Current implementation is per instance, eg. only some instances could be published on the farm from a single publish. This matches more future move to New Publisher.

For each remote publishable instance, separate full publish job is triggered, which might not be optimal. There should be a discussion of separating "extract" part (separate jobs) and final publish job (in similar fashion as currently renders are implemented.)

`MayaSubmitRemotePublishDeadline` is currently created, most of logic should be yanked out into Abstract class to allow similar implementation in a different host (eg. Houdini).

## Testing notes:
0. Deploy `openpype/modules/deadline/repository/custom/plugins/GlobalJobPreLoad.py` to `DL_repo/custom/plugins`
1. Create instance of Pointcache family
2. Set its Farm field to True
3. Publish
4. Check DL and eventually published folder
